app/src/main/java/com/github/campus_capture/bootcamp/fragments/ProfileFragment.java

```diff
@@ -80,16 +76,27 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     private void displayAlert(View view) {
         new AlertDialog.Builder(view.getContext())
                 .setTitle("Are you sure ?")
-                .setPositiveButton("Yes", new DialogInterface.OnClickListener() {
-                    @Override
-                     public void onClick(DialogInterface dialog, int which) {
-                        User.setSection(section);
-                        currentActivity.goToSignInFragment(emailText, passwordText, true);
-                    }
+                .setPositiveButton("Yes", (dialog, which) -> {
+                    User.setSection(section);
+                    storeUserOnDB(User.getUid(), section);
+                    storeUserOnDisk(User.getUid(), section);
+                    currentActivity.goToMainActivity();
                 })
                 .setNegativeButton("No", null)
                 .setMessage("Once selected, the selected section will be permanent ! Do you want to proceed ?")
                 .show();
     }
 
+    private void storeUserOnDB(String uid, Section section) {
+        FirebaseBackend backend = new FirebaseBackend();
+        backend.initUserInDB(uid, section);
+    }
+
+    private void storeUserOnDisk(String uid, Section section) {
```

Comment:
Is this cleared if you clear the app data...? Otherwise this would provide players with a work-around to change their section

Reply:
It is but since the section is stored on the DB and is fetched when the player login, there is no problem with that :-)